cpp/src/arrow/compute/kernels/vector_nested_test.cc

```diff
@@ -196,10 +238,47 @@ TEST(TestVectorNested, ListParentIndices) {
   auto tweaked = TweakValidityBit(input, 1, false);
   auto expected = ArrayFromJSON(int64(), "[0, 0, 0, 1, 1, 2, 2, 4, 4]");
   CheckVectorUnary("list_parent_indices", tweaked, expected);
+
+  {
+    // Construct a list-view with a non-empty null slot
+    auto input =
+        ArrayFromJSON(list_view(int16()), "[[0, null, 1], [0, 0], [2, 3], [], [4, 5]]");
+    auto tweaked = TweakValidityBit(input, 1, false);
+    auto expected = ArrayFromJSON(int64(), "[0, 0, 0, null, null, 2, 2, 4, 4]");
+    CheckVectorUnary("list_parent_indices", tweaked, expected);
+
+    // Swap some list-view entries
```

Comment:
What is the point of swapping exactly? Is it to make sure that the views are non-contiguous?

Reply:
> Is it to make sure that the views are non-contiguous?

Exactly.